chatclient/src/main/java/pl/sages/javadevpro/Client.java

```diff
@@ -36,15 +39,38 @@ public void sendMessage() {
             Scanner scanner = new Scanner(System.in);
             while (socket.isConnected()) {
                 String messageToSend = scanner.nextLine();
-                writer.write(messageToSend);
-                writer.newLine();
-                writer.flush();
+                String[] messageParts = messageToSend.split(" ");
```

Comment:
This splits also the filename, which leads to file upload failure. Suppose we should only split by first space, not each one.

Reply:
agree, change made